File: Find_Center_of_Star_Graph.go
package main

import (
	"fmt"
)

func main() {
	edges := [][]int{{1, 2}, {2, 3}, {4, 2}}
	fmt.Println(findCenter(edges))
}

func findCenter(edges [][]int) int {
	s1 := make(map[int]bool)
	// for i := 0; i < len(edges); i++ {
	// 	u, v := edges[0], edges[1]
	// 	fmt.Println(u, v)
	// }

	for _, i := range edges {
		u, v := i[0], i[1]
		if s1[u] {
			return u
		} else if s1[v] {
			return v
		}
		s1[u] = true
		s1[v] = true

	}

	return 0
}

File: Find_Center_of_Star_Graph/Find_Center_of_Star_Graph.py
from collections import defaultdict
def findCenter(edges):
    matrix = defaultdict(list)
    for edge in edges:
        u = edge[0]
        v = edge[1]
        matrix[u].append(v)
        matrix[v].append(u)

    l = len(matrix)-1
    print(matrix,l)
    for key in matrix.keys():
        if len(matrix[key]) == l:
            return key




# edges = [[1,2],[2,3],[4,2]]
edges = [[1,2],[5,1],[1,3],[1,4]]
print(findCenter(edges))